iroha/src/queue.rs

```diff
@@ -75,38 +75,31 @@ impl Queue {
                 .pending_tx_by_hash
                 .get(&transaction_hash)
                 .expect("Failed to get tx by hash.");
+
+            let mut bad_transaction = true;
+
             if !transaction.is_expired(self.transaction_time_to_live)
                 && !transaction.is_in_blockchain(world_state_view)
             {
                 if let Ok(signature_condition_passed) =
                     transaction.check_signature_condition(world_state_view)
                 {
-                    if is_leader {
-                        if signature_condition_passed {
-                            output_transactions.push(
-                                self.pending_tx_by_hash
-                                    .get(&transaction_hash)
-                                    .expect("Failed to get tx by hash. The map should contain txs that are in a queue.")
-                                    .clone(),
-                            );
-                            counter -= 1;
-                        }
-                    } else {
-                        output_transactions.push(
-                            self.pending_tx_by_hash
-                                .get(&transaction_hash)
-                                .expect("Failed to get tx by hash. The map should contain txs that are in a queue.")
-                                .clone(),
-                        );
```

Comment:
This part seems to break the logic here and decreases readability to my mind better to leave it like it was. I am going to add some comments here to describe what is happening better.